packages/relay/src/lib/eth.ts

```diff
@@ -1647,22 +1648,23 @@ export class EthImpl implements Eth {
     const transactionObjects: Transaction[] = [];
     const transactionHashes: string[] = [];
 
-    for (const result of contractResults) {
-      // depending on stage of contract execution revert the result.to value may be null
-      if (!_.isNil(result.to)) {
-        if(showDetails) {
-          // check the size of the block before querying for transaction details
-          if (contractResults.length >= this.ethGetTransactionCountMaxBlockRange) {
-            throw predefined.MAX_BLOCK_SIZE(blockResponse.count);
-          }
+    if (showDetails && contractResults.length >= this.ethGetTransactionCountMaxBlockRange) {
+      throw predefined.MAX_BLOCK_SIZE(blockResponse.count);
+    }
 
-          const transaction = await this.getTransactionFromContractResult(result.to, result.timestamp, requestId);
-          if (transaction !== null) {
-            transactionObjects.push(transaction);
+    for (let i = 0; i < contractResults.length; i+= this.ethGetBlockByResultsBatchSize) {
+      if (showDetails) {
+        Promise.all(contractResults.slice(i, i + this.ethGetBlockByResultsBatchSize).map(async result => {
+          // depending on stage of contract execution revert the result.to value may be null
+          if (!_.isNil(result.to)) {
```

Comment:
```suggestion
          if (result.to != null) {
```

This should be equivalent, does not uses `_` and is easier to understand.

Reply:
Existing code but will update

Reply:
Updated